stellar-multitenant/pkg/serve/serve.go

```diff
@@ -53,10 +58,17 @@ func (opts *ServeOptions) SetupDependencies() error {
 	opts.dbConnectionPool = dbConnectionPool
 
 	opts.tenantManager = tenant.NewManager(tenant.WithDatabase(dbConnectionPool))
+
+	signSvc, err := dependencyinjection.NewSignatureService(context.Background(), opts.SignatureServiceOptions)
+	if err != nil {
+		return fmt.Errorf("creating signature service: %w", err)
+	}
+
```

Comment:
🧵 This place is not the best to do dependency injection, since serve.go is already part of the application.

Here's a good definition of dependency injection:
> In software engineering, dependency injection is a programming technique in which an object or function receives other objects or functions that it requires, as opposed to creating them internally. ([Wikipedia](https://en.wikipedia.org/wiki/Dependency_injection#:~:text=In%20software%20engineering%2C%20dependency%20injection,opposed%20to%20creating%20them%20internally.))

If you do a project search, you will realize that we only import the dependencyinjection package from within the `cmd` package.